compiler/rustc_traits/src/chalk/db.rs

```diff
@@ -216,21 +216,21 @@ impl<'tcx> chalk_solve::RustIrDatabase<RustInterner<'tcx>> for RustIrDatabase<'t
             c: adt_def.repr().c(),
             packed: adt_def.repr().packed(),
             int: adt_def.repr().int.map(|i| match i {
-                attr::IntType::SignedInt(ty) => match ty {
```

Comment:
Any specific reason to make these changes?
Doesn't `rustc_attr` still reexport these types?

Reply:
Same applies to many changes in other files.
It may make sense to "normalize" reexports, but probably not in this PR, it's already too large.
Could you only move the necessary code to `rustc_abi` (replace the moved code with reexports where possible), and avoid any other changes?

Reply:
> Any specific reason to make these changes?

Code which computes layouts heavily relies on `ReprOptions`, so I moved `ReprOptions` to the `rustc_abi`. But its `int` field was `attr::IntType`, and `attr::IntType` depends on `ast::IntTy`. I tried to not change `attr::IntType` and instead added a `rustc_abi::IntegerType` for the same propose but it has a slightly different structure and uses `rustc_abi::Integer` instead of ast types. So (unlike most of the other changes) `attr::IntType` is not a reexport, it's a different type (and it exists in it's original shape, making it reexport will cause more changes).

> It may make sense to "normalize" reexports, but probably not in this PR, it's already too large.

:+1: 

> Could you only move the necessary code to rustc_abi (replace the moved code with reexports where possible), and avoid any other changes?

My intention was to do exactly this. I can fix them if you mention the cases.